### DISCUSSION TITLE
do not create a scrollable code block for more than 18 lines of code for reveal slides

### DISCUSSION THREAD

**Discussion by caimiao0714** [quarto-dev/quarto-cli#2910] [Q&A]
Hi all,

I've been learning quarto slides using reveal.js, and so far the experience has been great. The only thing that irritates me is when my code block is more than 18 lines, quarto just automatically creates a scrollable code block for me. However, I think it can just safely print out all the code because there is plenty of white space. Below are some code to reproduce the issue I have.

Is there any trick that I could use to tackle this issue of automatically creating scrollable code blocks?

Thanks,
Miao

```
format: 
  revealjs:
    width: 1920
    height: 1080

## Plot

::: columns
::: {.column width="65%"}
```{r}
#| echo: TRUE
#| eval: FALSE
ggplot() +
  geom_sf(data = gz_square_sf_count, 
          aes(fill = count), 
          color = 'gray80', size = 0.15) +
  geom_sf(data = CN_county,
          fill = NA, color = 'gray40', size = 0.3) +
  geom_sf(data = CN_city,
          fill = NA, color = 'gray50', size = 0.5) +
  scale_fill_continuous(
    type = "viridis", 
    option = 'A', 
    direction = -1, 
    limits = c(0, 120), 
    oob = scales::squish,
    breaks = seq(0, 120, 20), 
    labels = ~ ifelse(.x < 120, .x, '>120'),
    guide = guide_colorsteps(
      title = 'Count',
      title.position = "top",
      title.hjust = 0.5,
      label.hjust = 1,
      show.limits = TRUE,
      barwidth = unit(1, "cm"),
      barheight = unit(6, "cm"))) +
  labs(fill = 'count') +
  theme_void(base_size = 16)
```
```
:::

::: {.column width="35%"}
some text
:::
:::
```

![image](https://user-images.githubusercontent.com/20736990/196452144-720e0a13-1dc5-41bc-a07f-467e7fca4ea2.png)

**Comment by paddytobias** [comment]
Hey @caimiao0714 , I had the same issue. If you don't want the vertical scrolling you can add some css at the top of your Quarto script to overwrite the default


``````markdown
```{css, echo=FALSE}
.reveal code {
  max-height: 100% !important;
}
```
``````

**Reply by cderv** [reply]
Yes this is the rule that creates the scroll 
https://github.com/quarto-dev/quarto-cli/blob/f9bb1ebf6a5b0b879a5f885f16b84ff924fba277/src/resources/formats/revealjs/quarto.scss#L247-L255

`max-height: 500px` is set by Quarto. This is a default that works in most situation. 

Currently it is not configurable, but you can change it to your liking as shown above with a custom CSS or theme.  

Using a `css` chunk like above will only work for **knitr** engine, not Jupyter.

**Comment by sorenwacker** [comment]
Scrollable codeblocks destroy scrollable pages with revealjs.

**Reply by mcanouil** [reply]
@sorenwacker Please, be constructive. This kind of comments are not helpful. Thank you!
